js/renderTestimonials.js

```diff
@@ -43,94 +55,242 @@ function renderCards(data){
         if (i < 0) {iteration = size+i;}
         else if (i >= size) {iteration = i - size}
         // const iteration = i < 0 ? size-i-1 : (i>=size ? i-size : i);
-        let rate = '';
-        for (let j = 0; j < data[iteration].rating; j++) {
-            rate += '<i class="fa fa-star-o" aria-hidden="true"></i>';
-        }
-        HTML += `<div class="testimonial" style="width: ${width}px;">
-                            <div class="card">
-                                <div class="person"><img src="./img/${data[iteration].picture}.jpg" alt="persons picture"></div>
-                                <h4>${data[iteration].name}</h4>
-                                <p>${data[iteration].content}</p>
-                                <div class="rating">${rate}</div>
-                            </div>
-                        </div>`;
+        HTML += renderSingleCard(data[iteration], width);
```

Comment:
Jeigu kintamieji `i` ir `iteration` yra lygūs vienas kitam, tai gal turėtų likti tik vienas iš jų?

Reply:
Būtų variantas. Bet šitoj vietoj renderina korteles ir jų renderina daugiau negu reikia. Tai realiai 'i' yra kiek ciklu prasuks, kad surenderintų korteles, o 'iteration' nurodo, į kurį duomenų objektą reikia žiūrėti.
Šitoj vietoj aš svarsčiau, ką daryti, ar kurti naują kintamąjį, ar po 'if' parašyti funkcijas (55 ir 56 eilutės), kaip yra 58 eilutėj.
Pasirinkau kintamąjį. Gal tik nebent reikėjo pakomentuoti.
Kaip manai, kas tada geriau?

Reply:
```javascript
function renderCards(data, DOM) {
    const size = data.length;
    const extraHiddenCards = 2;
    const width = getCardWidth();
    let HTML = '';
    for (let i = -extraHiddenCards; i < size + extraHiddenCards; i++) {
        if (i < 0) {
            HTML += renderSingleCard(data[size + i], width);
        } else if (i >= size) {
            HTML += renderSingleCard(data[i - size], width);
        } else {
            HTML += renderSingleCard(data[i], width);
        }
    }
    return DOM.innerHTML = HTML;
}
```

Reply:
Viena vertus, speliamų elementų nori turėti iš abiejų pusių;
Kita vertus dėka to, atskirai skaičiuoti elemento index'o nebereikia.

Reply:
Makes sense.
Reikalas tik tas, kad iš kairės ir iš dešinės yra skirtingas kiekis papildomų kortų.
Iš kairės daugiau nei vienos nereikia, nes 1) sugriūtų kitos funkcijos, nes jos rašytos turint omeny, kad pradžioje yra tik viena paslėpta korta, 2) event listener'ių funcjos parašytos taip, kad prireikus sugeneruoja naują kortelę iš kairės, ar dešinės  todėl daugiau papildomai tikrai nereikia kažko bet kada keisti. 
Manau tiks, jei pašalinsiu kintamąjį 'iteration' ir parašysiu 'if'ą' kaip pas tave.

Reply:
Išbandyk pilną kodą 🙂 Vis gerai veikia ir niekas nesugriūna.

Reply:
Bet ne ta pirma kortelė yra rodoma :)